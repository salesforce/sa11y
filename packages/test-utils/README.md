# `@sa11y/test-utils`

Private package providing test utilities for @sa11y packages

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->


- [Usage](#usage)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

## Usage

```javascript
import { beforeEachSetup, domWithA11yIssues, domWithNoA11yIssues } from '@sa11y/test-utils';

import { registerSa11yMatcher } from '@sa11y/jest';

beforeAll(() => {
    registerSa11yMatcher();
});

beforeEach(() => {
    beforeEachSetup();
});

describe('...', () => {
    it('...', async () => {
        document.body.innerHTML = domWithNoA11yIssues;
        await expect(document).toBeAccessible();

        document.body.innerHTML = domWithA11yIssues;
        await expect(document).not.toBeAccessible();
    });
});
```
