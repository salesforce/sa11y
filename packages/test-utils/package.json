{
    "name": "@sa11y/test-utils",
    "version": "0.2.0-beta.0",
    "private": true,
    "description": "Private package providing test utilities for @sa11y packages",
    "license": "BSD-3-Clause",
    "homepage": "https://github.com/salesforce/sa11y/tree/master/packages/test-utils#readme",
    "repository": {
        "type": "git",
        "url": "https://github.com/salesforce/sa11y.git",
        "directory": "packages/test-utils"
    },
    "main": "dist/index.js",
    "types": "dist/index.d.ts",
    "files": [
        "dist/**/*.js"
    ],
    "dependencies": {
        "@sa11y/common": "0.2.0-beta.0"
    }
}
