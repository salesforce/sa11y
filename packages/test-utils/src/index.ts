/*
 * Copyright (c) 2020, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: BSD-3-Clause
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

export {
    audioURL,
    exceptionList,
    domWithA11yIssues,
    domWithDescendancyA11yIssues,
    domWithA11yIssuesBodyID,
    domWithNoA11yIssues,
    domWithNoA11yIssuesChildCount,
    domWithVisualA11yIssues,
    htmlFileWithA11yIssues,
    htmlFileWithNoA11yIssues,
    htmlFileWithVisualA11yIssues,
    a11yIssuesCount,
    a11yIssuesCountFiltered,
    shadowDomID,
    videoURL,
    customRulesFilePath,
    domWithA11yCustomIssues,
} from './test-data';
export { beforeEachSetup, cartesianProduct } from './utils';
