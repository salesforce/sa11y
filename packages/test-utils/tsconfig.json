{
    "extends": "../../tsconfig.common.json",
    "compilerOptions": {
        "rootDir": "src",
        "outDir": "dist"
    },
    "include": ["src"],
    "references": [{ "path": "../common" }]
}
