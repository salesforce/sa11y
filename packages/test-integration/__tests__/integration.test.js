/*
 * Copyright (c) 2020, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: BSD-3-Clause
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

import { beforeEachSetup, domWithA11yIssues, domWithNoA11yIssues } from '@sa11y/test-utils';

beforeEach(beforeEachSetup);

describe('integration test @sa11y/jest', () => {
    it('should have a11y matchers working with setup in jest.config.js', async () => {
        document.body.innerHTML = domWithNoA11yIssues;
        expect(expect.toBeAccessible).toBeDefined();
        await expect(document).toBeAccessible();
    });

    it('show throw error for inaccessible dom', async () => {
        expect.assertions(2);
        document.body.innerHTML = domWithA11yIssues;
        await expect(document)
            .toBeAccessible()
            .catch((e) => expect(e).toMatchSnapshot());
    });
});
