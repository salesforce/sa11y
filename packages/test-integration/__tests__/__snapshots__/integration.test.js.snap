// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration test @sa11y/jest show throw error for inaccessible dom 1`] = `
[Error: [2mexpect([22m[31m5 issues[39m[2m)toBeAccessible: expected document to have no accessibility violations but found 5 issues: [22m
[2m[22m
[2m [31m"â­• (bypass) Page must have means to bypass repeated blocks: html"[39m[22m
[2m	ðŸ”— Help URL: https://dequeuniversity.com/rules/axe/3.5/bypass[22m
[2m[22m
[2m[31m"â­• (document-title) Documents must have <title> element to aid in navigation: html"[39m[22m
[2m	ðŸ”— Help URL: https://dequeuniversity.com/rules/axe/3.5/document-title[22m
[2m[22m
[2m[31m"â­• (link-name) Links must have discernible text: a"[39m[22m
[2m	ðŸ”— Help URL: https://dequeuniversity.com/rules/axe/3.5/link-name[22m
[2m[22m
[2m[31m"â­• (landmark-one-main) Document must have one main landmark: html"[39m[22m
[2m	ðŸ”— Help URL: https://dequeuniversity.com/rules/axe/3.5/landmark-one-main[22m
[2m[22m
[2m[31m"â­• (page-has-heading-one) Page must contain a level-one heading: html"[39m[22m
[2m	ðŸ”— Help URL: https://dequeuniversity.com/rules/axe/3.5/page-has-heading-one([22m[32m0 issues[39m[2m)[22m]
`;
